resources/fetchers/ecr_factory.go

```diff
@@ -36,19 +38,15 @@ const (
 )
 
 func init() {
-	manager.Factories.ListFetcherFactory(ECRType, &ECRFactory{extraElements: getEcrExtraElements})
+	manager.Factories.ListFetcherFactory(ECRType, &ECRFactory{
+		KubernetesProvider: providers.KubernetesProvider{},
+		IdentityProvider:   awslib.GetIdentityClient,
+	})
 }
 
 type ECRFactory struct {
-	extraElements func() (ecrExtraElements, error)
-}
-
-type ecrExtraElements struct {
-	awsConfig               awslib.Config
-	kubernetesClientGetter  providers.KubernetesClientGetter
-	identityProviderGetter  awslib.IdentityProviderGetter
-	ecrPrivateRepoDescriber awslib.EcrRepositoryDescriber
-	ecrPublicRepoDescriber  awslib.EcrRepositoryDescriber
+	KubernetesProvider providers.KubernetesClientGetter
```

Comment:
why do we need a provider for this? for mocks?

Reply:
To be able to test the factory without creating a real Kubernetes client.